client/actions/userActionCreators.js

```diff
@@ -0,0 +1,131 @@
+import actions from '../constants/actions.js';
+
+export const checkSessionActionCreator = () => (dispatch) => {
+/*
+  thunked action creator that sends request to check
```

Comment:
what is thunked

Reply:
thunking (in general) refers to delaying the execution of a function. In this case, it's specifically for use with the 'redux-thunk' middleware (which is an installed package).

redux-thunk allows us to use async operations in our action creators. This allows us make a HTTP request before we dispatch an action, for example. (without thunking, there is no way to do an async operation before dispatching an action)